src/main/java/de/tum/cit/aet/artemis/core/web/CourseResource.java

```diff
@@ -1453,4 +1455,23 @@ public ResponseEntity<Long> getNumberOfAllowedComplaintsInCourse(@PathVariable L
         long unacceptedComplaints = complaintService.countUnacceptedComplaintsByParticipantAndCourseId(participant, courseId);
         return ResponseEntity.ok(Math.max(complaintService.getMaxComplaintsPerParticipant(course, participant) - unacceptedComplaints, 0));
     }
+
+    @GetMapping("courses/{courseId}/existing-exercise-details")
```

Comment:
Can you add JavaDoc here?
Please also make clear that the size of the returned arrays/sets does not have to be equal to the total amount of exercises